src/components/Timeline/Playbook/TextFilter.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import { TextField, FlatButton } from 'material-ui'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import * as filterActions from 'actions/filterActions'
+
+class TextFilter extends Component {
+  static propTypes = {
+    history: PropTypes.object.isRequired,
+    filters: PropTypes.object
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      value: ''
+    }
+  }
+
+  render () {
+    return (
+      <div>
+        <br />
+        <TextField
+          hintText='Search'
+          id='search-field-value'
+          value={this.state.value}
+          onChange={(event) => this.handleChange(event)}
+                />
+        <FlatButton
+          label='Search'
+          primary
+          onClick={() => this.submitSearch()}
```

Comment:
This component could be stateless if we separated the filter value update actions from the event fetch and url update actions; the search-field-value textfield could dispatch an action that updates the DataSearch portion of the filter reducer, and the submitSearch() onClick could be replaced with a different action that updates the history and starts the event fetch. I would go so far as to say that the filter submit button shouldn't be part of the same component as the TextFilter component or the TimeAndDatePicker component; it should be a sibling to those components. As it stands, datetime and text filters are submitted separately, which may not be intuitive to users, who may expect a workflow where they set all desired form values and then submit them all at once.

At this point, it's clear that the Filter reducer needs to be broken up into several reducers with a combineReducer statement.